.github/workflows/python-tests.yml

```diff
@@ -0,0 +1,28 @@
+name: Run Python Tests
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - src/config/**
+      - src/**.py
+      - src/requirements.txt
+jobs:
+  tests:
+    name: Run pytest
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.7'
+      - name: Install Requirements
+        run: |
+          python -m pip install --upgrade pip
+          cd src
+          pip install -r requirements.txt
+      - name: Run pytest
+        run: |
+          cd src
```

Comment:
I am not too sure if we need to change the working directory again as it was done in the previous step. If the directory context state is not preserved across steps then this will be needed, but I am don't know how the system behaves. You might want to test it somewhere else in a small sample workflow to confirm it.

Reply:
I tested it and we do need to do this. Alternative is to merge into same step but quite like them being separate so can see exactly which step fails more easily.